Mathlib/Probability/Variance.lean

```diff
@@ -0,0 +1,383 @@
+/-
+Copyright (c) 2022 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel, Kexing Ying
+
+! This file was ported from Lean 3 source module probability.variance
+! leanprover-community/mathlib commit f0c8bf9245297a541f468be517f1bde6195105e9
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Probability.Notation
+import Mathlib.Probability.Integration
+import Mathlib.MeasureTheory.Function.L2Space
+
+/-!
+# Variance of random variables
+
+We define the variance of a real-valued random variable as `Var[X] = 𝔼[(X - 𝔼[X])^2]` (in the
+`ProbabilityTheory` locale).
+
+## Main definitions
+
+* `ProbabilityTheory.evariance`: the variance of a real-valued random variable as a extended
+  non-negative real.
+* `ProbabilityTheory.variance`: the variance of a real-valued random variable as a real number.
+
+## Main results
+
+* `ProbabilityTheory.variance_le_expectation_sq`: the inequality `Var[X] ≤ 𝔼[X^2]`.
+* `ProbabilityTheory.meas_ge_le_variance_div_sq`: Chebyshev's inequality, i.e.,
+      `ℙ {ω | c ≤ |X ω - 𝔼[X]|} ≤ ENNReal.ofReal (Var[X] / c ^ 2)`.
+* `ProbabilityTheory.meas_ge_le_evariance_div_sq`: Chebyshev's inequality formulated with
+  `evariance` without requiring the random variables to be L².
+* `ProbabilityTheory.IndepFun.variance_add`: the variance of the sum of two independent
+  random variables is the sum of the variances.
+* `ProbabilityTheory.IndepFun.variance_sum`: the variance of a finite sum of pairwise
+  independent random variables is the sum of the variances.
+-/
+
+
+open MeasureTheory Filter Finset
+
+noncomputable section
+
+open scoped BigOperators MeasureTheory ProbabilityTheory ENNReal NNReal
+
+namespace ProbabilityTheory
+
+local macro_rules | `($x ^ $y) => `(HPow.hPow $x $y) -- Porting note: See issue #2220
+
+-- Porting note: this lemma replaces `ENNReal.toReal_bit0`, which does not exist in Lean 4
+private lemma coe : ENNReal.toReal 2 = (2 : ℝ) := rfl
+
+/-- The `ℝ≥0∞`-valued variance of a real-valued random variable defined as the Lebesgue integral of
+`(X - 𝔼[X])^2`. -/
+def evariance {Ω : Type _} {_ : MeasurableSpace Ω} (X : Ω → ℝ) (μ : Measure Ω) : ℝ≥0∞ :=
+  ∫⁻ ω, (‖X ω - μ[X]‖₊ : ℝ≥0∞) ^ 2 ∂μ
+#align probability_theory.evariance ProbabilityTheory.evariance
```

Comment:
In https://github.com/leanprover-community/mathlib4/pull/4824 we went with the name `eVariation`. Should this be `eVariance` for consistency? Not sure.

Reply:
I slightly prefer `evariance` because I already saw names like `edist` and `aestronglyMeasurable`. Anyways, I added a porting note.